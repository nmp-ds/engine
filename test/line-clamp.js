import { setup } from "./_helpers.js";
import { describe, expect, test } from "vitest";
import { getNumericArrayInRange } from '#utils';


setup();

describe('line-clamp', () => {
  test('should render styles for static classes', async ({ uno }) => {
    const staticClasses = getNumericArrayInRange(1, 9).map(n => `line-clamp-${n}`);

    const { css } = await uno.generate(staticClasses);
    expect(css).toMatchSnapshot();
  });
  test('should not render styles for invalid classes', async ({ uno }) => {
    const arbitraryClasses = ['line-clamp', 'line-clamp-foo'];

    const { css } = await uno.generate(arbitraryClasses);
    expect(css).toMatchInlineSnapshot('""');
  });
});

