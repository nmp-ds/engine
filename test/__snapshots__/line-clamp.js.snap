// Vitest Snapshot v1

exports[`line-clamp > should render styles for static classes 1`] = `
"/* layer: default */
.line-clamp-1{overflow:hidden;display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:1;line-clamp:1;}
.line-clamp-2{overflow:hidden;display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:2;line-clamp:2;}
.line-clamp-3{overflow:hidden;display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:3;line-clamp:3;}
.line-clamp-4{overflow:hidden;display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:4;line-clamp:4;}
.line-clamp-5{overflow:hidden;display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:5;line-clamp:5;}
.line-clamp-6{overflow:hidden;display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:6;line-clamp:6;}
.line-clamp-7{overflow:hidden;display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:7;line-clamp:7;}
.line-clamp-8{overflow:hidden;display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:8;line-clamp:8;}
.line-clamp-9{overflow:hidden;display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:9;line-clamp:9;}"
`;
